services/awp/onboarding_common.go

```diff
@@ -61,64 +61,82 @@ type GetAWPOnboardingResponse struct {
 
 // Common functionality
 
-func CreateAWPOnboarding(client *client.Client, req interface{}, path string, queryParams CreateOptions) (*http.Response, error) {
-	// Define the maximum number of retries and the interval between retries
-	maxRetries := 3
-	retryInterval := time.Second * 5
-
-	// Initialize the response and error variables outside the loop
+func retryRequest(f func() (*http.Response, error), maxRetries int, retryInterval time.Duration) (*http.Response, error) {
 	var resp *http.Response
 	var err error
 
-	// Attempt the request up to maxRetries times
+
 	for i := 0; i < maxRetries; i++ {
-		// Make the request
-		resp, err = client.NewRequestDo("POST", path, queryParams, req, nil)
+		resp, err = f()
 		if err == nil {
-			// If the request was successful, return the response
+	
 			return resp, nil
 		}
 
-		// If the request failed with a 404 status code, wait for the retry interval before trying again
-		if resp != nil && resp.StatusCode == 404 {
+		if resp != nil && (resp.StatusCode >= 400 && resp.StatusCode < 500) {
 			time.Sleep(retryInterval)
 		} else {
-			// If the status code is not 404, return the response and error immediately
+		
```

Comment:
better to 'break' from the loop instead of return

--> The advantage is that in case of error, we will see the same msg for all cases...

Reply:
done